### DISCUSSION TITLE
Suggestion to allow greater useful ship diversity without adding new assets: "Internals Refit" any ship.

### DISCUSSION THREAD

**Discussion by DoublePants** [endless-sky/endless-sky#10277] [Ideas]
There are so many cool looking ships that eventually just become obsolete, so you end up using the same few ships every playthrough. My suggestion would be to allow any ship to be brought to various allied factions to undergo an "internals refit" to make the ship's base stats more similar to those of whichever faction you brought it to.

For example, my favorite ship looks-wise is the Saber. Unfortunately it's pretty much crap by the time you can get one. What I propose is that, if I have a Saber, I can bring it to the Hai homeworld (if my reputation with them is high enough of course) and "refit" it to have base stats similar to a Sea Scorpion. Or I could bring it to the Remnant and "refit" it to have stats similar to an Ibis (including the built-in shield regeneration, hull repair, sensors, and ramscoop). The refit would of course not change the number of weapon mounts or escort bays since that would mess with animations.

I think most of the infrastructure to do this is already in the game (since you can change the stats on individual ships), and the main things needed would be prices to do such refits and which ships can refit into which sets of stats. Obviously a Saber shouldn't be able to refit into an Albatross. I'd suggest that each category of ships which includes a size (Light/Medium/Heavy) or is inherently a size (Interceptor, Fighter, Drone) be eligible for an internals refit, and there should just be a set of base stats for each race that can become the player's ally for each of those categories that any ship in those categories can switch to.

**Comment by TheGiraffe3** [comment]
The problem is technically then you could refit your Modified Argosy into a Model 32. Or your Hauler II into a Deep River. So I'm not sure that would work.

**Reply by DoublePants** [reply]
Which faction would you get a Kor Mereti refit from?

Do you think it would have a significant negative impact on the game if a Hauler II had a Deep River's base stats, but only for the player?

**Reply by TheGiraffe3** [reply]
1). The Kor Mereti, after the Wanderer campaign. 
2). Saugia seems to think so, and I am willing to guess other people will too.

**Comment by Saugia** [comment]
This is unlikely; ships are intentionally supposed to be either weak or strong, and it's up to you to re-outfit them with new tech and such to make them better. The Saber especially is meant to be a weak old ship in human history; that was the entire reason it was added. For a vanilla experience, we don't want to add outfits or retools that would otherwise heavily change the stats of a ship. This also would be incredibly difficult to balance; the entirety of the game is balanced around ships having the stats that they do, and not around the existence of outfits that can drastically update base stats (outside of things such as cargo expansions, outfit expansions, etc..)

This is however, something commonly done with plug-ins.

**Reply by TheGiraffe3** [reply]
@DoublePants was wanting to be able to have the stats of one ship but the sprite of another ship without save editing, I believe.

It wouldn't be terrible if you paid the amount of money that you would to buy the ship you're refitting to.

**Reply by Saugia** [reply]
Yeah, I know. It's the same thing. We wouldn't add a vanilla outfit that would change the base stats of a ship across the board, let alone anything significant, or a sprite change. This is essentially what save-editing or plug-ins are for; not vanilla.

Again, the best example is outfits that would increase health pools. In general, any outfit that increases a stat beyond the maximum is already a balance concern; cargo and outfit expansions have been core to this game as they were in EV, but doing so with especially weapon space, shields, health, and other aspects like these is a huge balance headache, even if we were to find an "average ground" for them.

Even within the same tier or within a ship class, or with monetary balance, any outfits that blanket increase stats are too much for something to have for vanilla. We also generally avoid adding outfits specifically only the player would use; if the AI were to use an outfit like this, it would still be a nightmare for balance. So, as with times past, we have turned down requests for stat-increasing outfits outside of what already exists. Perhaps there may be very rare future ones, perhaps an alien race adds a few shields or hull with a limited outfit, but nothing like general stat adjustments.

**Reply by TheGiraffe3** [reply]
I'm not sure you get what we're saying here.

_It is as if_ you had bought a new ship, except for the fact that it has the sprite of a different ship.

**Reply by Saugia** [reply]
The discussion suggests:

undergo an "internals refit" to make the ship's base stats more similar to those of whichever faction you brought it to.

This reads, "change base stats". That is what I'm talking about. I don't see anything about a different sprite. The request is to have an outfit that, if you were to bring a given ship to a given faction, said faction's outfit would upgrade the base stats of the ship more closer to that faction's tier/stats. Which is something we wouldn't really do for above and past reasons.

**Reply by TheGiraffe3** [reply]
I'm saying that from a mechanics perspective it would _seem_ as if you bought a new ship. The only difference is that it has a different sprite.

From a player's perspective, their ship got new stats.

**Reply by Saugia** [reply]
It would seem that way, yes, but ultimately, you are changing the base stats of a ship, which is what we wouldn't do.

I disagree that it's viewed as a different sprite. You're taking a ship and updating its stats to another faction's. You are not taking another faction's ship and changing the sprite to a lower tier ship. In either case, this is something best manipulated with plug-ins, not vanilla outfits/mechanics.

**Reply by TheGiraffe3** [reply]
Outfits?

But fine, though I would like to hear the OP's arguments.

**Reply by Quantumshark** [reply]
I would also make the point that the number and type of hardpoints a ship has is both a) a vital part of balancing the ship, and b) tied to the ship's sprite.

If you can get the hardpoint count of ship A by using ship A's sprite, while using the other stats of ship B, you can use this to create a ship that combines, perhaps, a desirable hardpoint count on ship A, which has other downsides to balance this hardpoint count, with the desirable stats from ship B, which are perhaps a benefit which is supposed to come with the tradeoff of low hardpoint count. With that in mind, this idea creates a huge balance issue.

Beyond that, though, I believe ships' stats should be connected to the ship, rather than being interchangeable. Ships lose their uniqueness if you can give any ship the stats of any other.

**Reply by DoublePants** [reply]
If you had read my suggestion completely you would see that I specifically mentioned that it would not change the number and type of hardpoints.

**Reply by TheGiraffe3** [reply]
But then, unfortunately, the sprite doesn't fit.

**Reply by DoublePants** [reply]
In response to: "Ships lose their uniqueness if you can give any ship the stats of any other."

The issue is that ships already lose their uniqueness past certain points in the game for the player. There's no functional difference between a Berserker and a Fury once you have a medium warship. They're both flies to be swatted and you'd never have any desire to use one ever again. There's no functional difference between a Falcon and a Vanguard once you have a Remnant ship and you'd never have any desire to use one ever again.

It's going to become an even bigger issue if Gegno ships ever become obtainable in regular gameplay. There's no reason to ever use any of them because there's no outfit that increases shield strength. 

It just seems odd to me to spend all this time making so many cool and unique ships only to have there be no reason for the player to ever use 90% of them.

**Reply by Saugia** [reply]
That's the whole point. Gegno ships, human ships, especially the older ones, are added to the game to represent that disparity in usefulness. The goal of Endless Sky is not to have everything be useful to the player at all times, it's to tell stories with varying states of existences/uses. I've intentionally added the Gegno as a species a weaker tier than humans, and I intentionally don't want them to be better, as there was no representation of that prior to their existence.

That was the same intention and reasoning for older human ships such as the Saber - they're meant to add to the lore in terms of older human ships that no longer have use. The value comes in its uniqueness and looks, not the ability to make it relevant again in the game's current timeline. It's not about cool ships being useful, it's about the lore that came with them.

Human ships/outfits and items in general were never meant to be relevant for the entire game; each area and campaign section has its own usefulness. We don't want to make everything in the game relevant at every point, or provide ways to make them relevant, beyond what you can already do in-game (outfitting).

That said, years of history regarding these suggestions have been given the response that they aren't something desirable for vanilla content for many balance reasons, and Devs have felt this is what makes the best sense for vanilla. I don't see this decision changing as the balance concerns still outweigh the utility, and gameplay-wise not everything is supposed to be relevant or able to be made relevant.

That said, it is a perfect case to use plug-in content, and, plug-ins should be seen as a great bridge for content that may not make sense in what we envision for vanilla. MZ himself said plug-ins are a huge part of ES; I feel more people should use them and appreciate them for things that don't fall in line with vanilla, and we have a great community who appreciates them.

**Reply by TheGiraffe3** [reply]
Not useful... until we get a Gegno start.

**Reply by Quantumshark** [reply]
> If you had read my suggestion completely you would see that I specifically mentioned that it would not change the number and type of hardpoints.

That's exactly the problem I was referring to. You don't change the number and type of hardpoints, meaning the ship keeps the number and type of hardpoints associated with its sprite, _not_ the number and type associated with the stats it's gained, meaning it now has stats and hardpoints that weren't balanced for each other.

**Reply by DoublePants** [reply]
> > If you had read my suggestion completely you would see that I specifically mentioned that it would not change the number and type of hardpoints.
> 
> That's exactly the problem I was referring to. You don't change the number and type of hardpoints, meaning the ship keeps the number and type of hardpoints associated with its sprite, _not_ the number and type associated with the stats it's gained, meaning it now has stats and hardpoints that weren't balanced for each other.

That would be an issue if it weren't for the fact that the lower tech ships you would be upgrading pretty much always have fewer hardpoints, which would leave them to still generally be weaker than the ships you'd be upgrading them to be like.

>The goal of Endless Sky is not to have everything be useful to the player at all times, it's to tell stories with varying states of existences/uses.

I believe that goal would be enhanced by letting players personalize their own ships in an official capacity. Every Gegno ship in the game would still be weaker than human ships, except for the one piloted by the player. Do you really like the end-game consisting of only a handful of viable ships? Would you really not have more fun if you could upgrade ol' reliable or the Millennium Falcon or Serenity or the Event Horizon to continue to at least kind of compete into the end-game if you so chose?

I think there's a lot of storytelling room that's left on the floor by having such a large portion of the content become obsolete.  The outfits system is a great way to make a ship feel like it's your ship and not just another bucket of bolts, but as long as that outfit system doesn't extend to shields, hull, and other internals, you're still eventually going to have to park your favorite ship in a hangar somewhere and never use it again once you hit a certain point in the game, so there's no point in becoming too attached or wasting too many resources on any ship that isn't one of the half dozen or so viable endgame ships. When I end up in a Hurricane or a Kar Ik Vot 349 every game, those ships stop being interesting.

Either way though, I accept that the interest in a change like this doesn't seem to currently be widely-held. I do hope people reconsider it for the future, though, because I think it would make the game objectively better in the long-term.

**Comment by Saugia** [comment]
You can always outfit human ships with alien weapons or tech, and it gives them and even more fun purpose and unique build style if they're of different usefulness. The second you add the ability to upgrade ships across the board you instantly lose that kind of special personalization; I think what you believe to be personalization is not personalization, but relevance, and those are two very different aspects of gameplay. You can personalize any ship you want, weak or strong, right now. Changing stats is not personalization; it is purely changing the balance of a ship to be better in comparison to another ship. That's not personalization, that's power comparison and creep, and a balance concern. I've noted that most times balance concern has been brought up in this conversation it is either not addressed or not important enough to consider, but to us, it is extremely important as the game's balance is very fine-tuned and takes a look at the entire board.

The Millenium Falcon, Serenity, etc., are not good examples because in comparison would be like saying a Human ship fighting against Hai tech in their respective universes; there are notable jumps with technology when it comes to Korath and Wanderer content. You could upgrade your "Millenium Falcon" with alien outfits, so akin to giving it more advanced weapons in its universe, and you are doing as you say and personalizing a ship to make it more useful. What you aren't doing is upgrading every facet of the ship to simply be similar in numbers with its enemies; you're not buffing every part of your Millenium Falcon until it can take on a Star Destroyer. Again, that's not personalization, that's power creep.

For example, many players have fiddled with Gegno ships in testing and have come up with some very neat builds. Sure, they will never be getting shields, but that is all the more reason they're unique and special, and many of them work as decent ships. Adding a way to simply upgrade a ship takes away that uniqueness and just makes every ship bland and boring, and muddies the very intentional gaps between content. And yes, the outfit system is a great way to change your ship to be not just another ship, but *base stats* are not the same thing as *outfits/internals*. There is a clear difference here, and we intend to keep it that way. Again, there may be small exceptions here and there, or outfits that touch on some parts (and are appropriately balanced, such as outfit expansions), but anything across the board is too major of a balance concern for any consideration/essentially just stat-sticking/stat-checking.

I don't think a large portion of the game is being obsolete; and to me, that sounds like a large over-exaggeration. You could still do missions in older and weaker ships; you could have fleets of them. You could outfit them differently. The entire purpose in any game is to have varying degrees of ships and outfits - *both* weak and strong. The moment you make weak ships be able to compete with the strong, you break all sense of uniqueness and variance. These kinds of gaps are meant to be there. Gegno ships, older human ships, both were added to the game *specifically* for lore purposes and representation *of* weaker ships; not as alternate ships to compete with the rest in the game. If anything, there are far too many average ships and there are a decent amount of good ships, but in comparison not many weaker ships or bad ships. Human ships aren't meant to take on Quarg ships. Wanderer ships are meant to be used for the later parts of Wanderer Campaign. Yes, you do end up with some of the same stronger ships later in the game, but that is partially intentional. Arguably that makes for even more interesting gameplay for attempting in more fun ship builds or unique fleets, etc; all more interesting and rewarding than simply giving ship x more numbers, just because it doesn't have the same numbers.

Using your flagship and feeling your wasting time on it is subjective. A core gameplay aspect of Endless Sky *is* upgrading your flagship along the way, both with outfits, and changing the ship class. This is an intended game path, you're never meant to play the same ship for notably long, and the time you spend in the same ship is decently long in each relevant part of the campaign. That may feel short to you and some people, but it's the kind of timeframe we're going for and what we've worked with.

Ultimately, this seems to lead into a circular argument - we're happy with the direction we've chosen for the game and these types of aspects, and believe it's been the best for the game's balance and storytelling. You are more than welcome to disagree however, and that's why I strongly recommend this becoming plug-in usage. It hurts nobody, and it adds the ability for any interested, and avoids any balance concerns or vision differences with where development sees the game. Thank you for the suggestion though! At the end of the day, it's good to talk about it, but it's entirely normal in game development to make choices and move on.

**Reply by DoublePants** [reply]
>I've noted that most times balance concern has been brought up in this conversation it is either not addressed or not important enough to consider, but to us, it is extremely important as the game's balance is very fine-tuned and takes a look at the entire board.

Part of the reason I haven't addressed this is because it doesn't make sense to me and I'm not sure I have the necessary experience to state definitively that my sense is correct. What kind of balance issue might be created by increasing a ship's base stats that isn't equally present in simply buying a newer ship with better base stats?

> The Millenium Falcon, Serenity, etc., are not good examples because in comparison would be like saying a Human ship fighting against Hai tech in their respective universes; there are notable jumps with technology when it comes to Korath and Wanderer content. You could upgrade your "Millenium Falcon" with alien outfits, so akin to giving it more advanced weapons in its universe, and you are doing as you say and personalizing a ship to make it more useful. What you aren't doing is upgrading every facet of the ship to simply be similar in numbers with its enemies; you're not buffing every part of your Millenium Falcon until it can take on a Star Destroyer. Again, that's not personalization, that's power creep.

I'm not sure I agree with using the phrase "power creep" for this. It's kind of like calling buffing a weapon in Elden Ring "power creep;" you're making your existing tool stronger over the course of the game instead of acquiring new tools. It's not power creep, it's just player progression. Power creep usually refers to the introduction of new mechanics which invalidate old ones. So if you were to introduce a bunch of new human ships that just had better stats than the old ones but left the old ones in the game, that would be power creep. In fact, you could argue that the game's current progression system has a lot more in common with power creep than what I'm asking for; as you go through the game, new things are handed to you which render the old things obsolete. That's much more like power creep than being able to upgrade your existing ship to keep up with new ships that are made available.

> For example, many players have fiddled with Gegno ships in testing and have come up with some very neat builds. Sure, they will never be getting shields, but that is all the more reason they're unique and special, and many of them work as decent ships. Adding a way to simply upgrade a ship takes away that uniqueness and just makes every ship bland and boring, and muddies the very intentional gaps between content. And yes, the outfit system is a great way to change your ship to be not just another ship, but _base stats_ are not the same thing as _outfits/internals_. There is a clear difference here, and we intend to keep it that way. Again, there may be small exceptions here and there, or outfits that touch on some parts (and are appropriately balanced, such as outfit expansions), but anything across the board is too major of a balance concern for any consideration/essentially just stat-sticking/stat-checking.

But is it "unique" to have a ship that's strictly inferior to another ship? Does it make Remnant ships LESS unique because they DO have shields just like everyone else? I can kind of agree with one point here, wherein I think the IDEAL would be to have in-faction ship progression which follows the design cues of that faction. For example, in my ideal scenario there would be a Gegno "tier 3" ship that might still not have shields, but would have vastly more hull than current Gegno ships; so it still feels like a Gegno ship but you can actually use it without knowing for a fact that you're weakening yourself for aesthetic purposes. But that would take entirely new systems and stat sheets which is why I suggested this comparatively much more simple approach.

> I don't think a large portion of the game is being obsolete; and to me, that sounds like a large over-exaggeration. You could still do missions in older and weaker ships; you could have fleets of them. You could outfit them differently.

Sure, you COULD. And you could play the game one-handed, too. You could also STILL do that if a system were added to tier up existing ships; you wouldn't HAVE to tier ships up. But it's absolutely a fact that if you're concerned with playing optimally or even just in not intentionally handicapping yourself, there are only about a dozen ships you would ever reasonably use given the choice (setting aside the Pug Arfecta anyway, since if you don't there's really only one). Would it be neat to do a play-through where you limit yourself to only human interceptors? Yeah, for a while. Would it be equally neat to be able to upgrade a human interceptor so it has a reasonable durability and therefore you can play a competitive dogfighter style ship in the late game? Also yes. And adding that capacity wouldn't prevent you from being a swarm of tier 1 interceptors. It's effectively just letting you slap a Berserker skin over a Grasshopper.

>The entire purpose in any game is to have varying degrees of ships and outfits - _both_ weak and strong. The moment you make weak ships be able to compete with the strong, you break all sense of uniqueness and variance.

I think this would be true if I were asking for the tiers to be flattened, i.e. to make the base Quarg ship on the same level as a base human ship. But I'm not asking for that.

It sounds to me like you just enjoy the ship progression and are concerned that my idea would negatively impact that, but I'm not sure how it would. I think generally, Remnant ships would still be better than Gegno ships due to Remnant ships tending to have more hardpoints. You just wouldn't have to toss your Falcon in the trash the day you get your Ibis.

**Reply by Saugia** [reply]
>But it's absolutely a fact that if you're concerned with playing optimally or even just in not intentionally handicapping yourself, there are only about a dozen ships you would ever reasonably use given the choice

> Would it be neat to do a play-through where you limit yourself to only human interceptors? Yeah, for a while.

Yes, and this is intended behavior. This is an intentional design choice for gameplay. We don't want to make more ships viable through blanket means, and we don't want to make early game ships as viable as later ones. The viability comes in outfit customization, not balance changes to a ship's overall definition. Human ships are meant to be relevant in human content, alien ships are meant to be relevant in alien content. The relevance is localized to whatever part of the campaign you are in, not branching across the entire game. And as you say, you still can attempt it and get interesting or fun results.

For the sake of clarification, this is not just my personal take on ship progression. These beliefs have been held by developers over years with the same or similar kind of suggestions for the same concerns. TLDR, balance outweighs the utility, and you disagree what personalization is vs. power scaling compared to what the dev team agrees upon it being (which is totally understandable). There's not much else I can say besides this a core part of ES's design, and we're pretty happy with it, and a plug-in is more than suitable for adding this to the game should you desire it.

**Reply by DoublePants** [reply]
I note you did not answer my question about what kind of balance issues there would be were my suggestion implemented.

**Reply by Saugia** [reply]
I did; ships are intended to be balanced for certain parts of the campaign. Human ships are meant to be balanced for human content, alien ships are meant to be balanced around alien content. A human ship is not meant to be brought into another tier as a relevant contender; these gaps of power are an intentional game design point.

And when speaking of balance, I mean ship stats, hardpoints (or lack of change of hardpoints with a change of base stats), etc.. This is primarily where conversation comes from the past with shield/hull addon outfits- they are extremely difficult to balance around as you're adding stats to a ship that otherwise are never supposed to change outside of outfits. When you do so for *every* stat - capacities, health, speed, you're creating a balance problem where the ship now has stats it should not have and the game is not balanced around, and other ships are not balanced around. 

The player isn't supposed to have a grand advantage over the AI that the AI isn't using; outfitting is intentionally where this limit is handled. I think that the gravity of editing a ships' base stats is being underestimated; even the slightest tweaks is a domino that befalls the rest of the game.

@Quantumshark could speak better on specific balance concerns, regardless, our standpoint on this has been the same over years.

In any case, I digress- a plug-in is an amazing way to implement this, rather than continue to push it for vanilla when we already have made a decision based on it. We're talking in circles at this point. The current behavior is intentional, and the suggestion is to change that behavior. It's normal to come to disagreements of course, and nothing is stopping from anyone from creating a plug-in or a copy of ES's repo to include something that we feel does not fit vanilla. That's something that's great about open source projects.

**Reply by Quantumshark** [reply]
> I note you did not answer my question about what kind of balance issues there would be were my suggestion implemented.

I did answer your question, albeit admittedly before it had been asked. I'll clarify my point here.

A ship's number and type of hardpoints is an important factor in balancing it, and being able to switch out any ship's other stats for those of another ship while leaving the hardpoint count intact is a balance nightmare.
Rather than just balancing ships individually, we'd then need to ensure no combination of one ship's hardpoints and another ship's stats was unbalanced, which would really limit our ability to use hardpoint counts to vary ships.
To give a specific example of the kind of circumstance in which this would create issues, look at the Hawk and the Fury. The Hawk is tougher, faster, and has more outfit and weapon space than the Fury, but the Fury has three times more guns, so the two are balanced against each other.
With your proposal, you could 'internals refit' a Fury, give it the stats of a Hawk, and end up with a ship that had all the strengths and none of the weaknesses of both ships. And you could do similar things with ships in similar balance positions later in the game - giving the weaker but hardpoint-rich Albatross or Shield Beetle the stats of the strong but somewhat hardpoint-limited Derecho, is another example.

**Reply by DoublePants** [reply]
> To give a specific example of the kind of circumstance in which this would create issues, look at the Hawk and the Fury. The Hawk is tougher, faster, and has more outfit and weapon space than the Fury, but the Fury has three times more guns, so the two are balanced against each other. With your proposal, you could 'internals refit' a Fury, give it the stats of a Hawk, and end up with a ship that had all the strengths and none of the weaknesses of both ships.

This is not what I was suggesting.

**Reply by Saugia** [reply]
> to undergo an "internals refit" to make the ship's base stats more similar to those of whichever faction you brought it to.

This is what you're suggesting, and it will result in a similar situation. You'd be upgrading a Fury to, let's say, Remnant stats, and this is not taking into consideration of a Fury's hardpoint count to internal space count, and it's also not talking account of how Remnant ships (interceptors in this case) or any other alien ships balance around hardpoint and internal space counts (and health, and speed, and mass, and drag, and the list goes on).

We aren't talking about adding or removing hardpoint counts; we're saying any and all base stat changes (blanketed, not small fine-tuned outfits like the current expansions) break balance because ships' base stats are built tightly to their gun and turret amounts and vice versa, as well as to the balance of other ships in their specific faction, and then their faction in comparison to other factions - hence why I say the second you change any base stats, especially all at once, with the intent of matching another level/faction/etc., you're creating a domino effect of balance issues.

**Reply by Saugia** [reply]
Though, let's humor this. I have a Saber. The Saber is a sub-par and very old human ship design. It can play okay for small human content. It is not designed for human campaign content. Keeping aside the design intention that it is never meant to be good or useful in harder content, or later game content, I want to make it useful.

With your suggestion, let's say I want to make a Saber's base stats reflect the balance of the Korath. For a lack of a better way to explain the method, this is us effectively taking the Saber sprite, and balancing all of its stats based around its guns, weapons, and other Korath ships. We're essentially left with a Korath-tier Saber. However, what you've done isn't make a weak ship relevant in Korath content; you've essentially created an entirely new ship type that has been balanced around other Korath ships, and it just happens to look like a Saber. It's no longer balanced as a Saber, nor is it a personalized Saber, it is a new entity altogether, just with the same image as a Saber. The same applies if I wanted the Saber to be just as good as a Hai ship, Coalition ship, Gegno ship (so backwards), or even another human ship (since you could consider the older human ships are one level, and the rest of them are another).

That's one method of this - balancing within the given faction and to a given faction's ships, and is more meticulous. Another would be to simply add numbers to the Saber's stats until it can compete with Korath ships (or other examples). This falls into the balance concerns above because you're not balancing the ship based around its intentional role, hardpoint count, and sibling ships. You're adding stats for the sole purpose of competing with something with more stats. Again, this isn't personalization; this is power scaling.

I understand that this is your desire, but it doesn't suit what progression levels we do in vanilla. Ultimately, ships are meant to only be relevant for certain parts of the game, certain niches, or not at all, for a decent amount of things, or for many things, and all based around the lore of the ship, or what ships we expect the player to be using in parts of campaign content. Adding a function to make *any* ship do that (to vanilla; again, you can easily make this an add-on and have the functionality regardless) is out of this scope and goes against a lot of ships' design principles.

Some ships will always be weak. Some will always be strong. Some will always be average. And some may shift in these categories based on what outfits or playstyles used with them - *this* is that small boost that can change a ship to be slighty better. If you can't make a Saber as good as an Osprey, or as a Sea Scorpion, no matter how you outfit it, then it is never meant to compete with those ships. That's the kind of environment Endless Sky is based around.

(And this is not to say there won't ever be balance changes - some ships that are too strong can still get nerfs, and some ships that aren't strong enough can still get some love. For example, the recent mass changes, and possible future changes, are allowing smaller and faster ships to be able to compete with larger and now slower ships. A larger ship may be suffering from nerfs, and we can bring it up. What we aren't doing is trying to make a ship that is strictly inferior to another, especially across tiers or campaign content, be able to be just as relevant as that other ship.)

**Reply by DoublePants** [reply]
> With your suggestion, let's say I want to make a Saber's base stats reflect the balance of the Korath. For a lack of a better way to explain the method, this is us effectively taking the Saber sprite, and balancing all of its stats based around its guns, weapons, and other Korath ships. We're essentially left with a Korath-tier Saber. However, what you've done isn't make a weak ship relevant in Korath content; you've essentially created an entirely new ship type that has been balanced around other Korath ships, and it just happens to look like a Saber. It's no longer balanced as a Saber, nor is it a personalized Saber, it is a new entity altogether, just with the same image as a Saber. The same applies if I wanted the Saber to be just as good as a Hai ship, Coalition ship, Gegno ship (so backwards), or even another human ship (since you could consider the older human ships are one level, and the rest of them are another).

Alright, so which Korath ship would the resulting Saber be stronger than? 5 gun ports and 1 turret mount. Let's say it gets Model 128 stats, another medium warship. It has 2 gun ports and 5 turret slots. Not only are turret slots generally more valuable, but you'd still have fewer total weapons even if you put equal value on turrets and guns. So the Model 128 wins

What about a Tek Far 78? 4 gun ports, 5 turret slots versus the Saber's 5/1, and that's even ignoring the 9 fighter bays. Tek Far wins. By a lot.

What about a Tempest? 4/3 against the Saber's 5/1.

The Bunrodea Kama makes it look like a joke with its 10/1.

The only ship I could find where the resulting Saber would be better is in comparison to a Pug Enfolta, which is 4/1 vs the Saber's 5/1, but by the time you might be able to ally the Pug if such missions were added, the resulting ship would still be pathetically weak compared to just doing the exact same thing with a Tempest, and my impression of the Pug Enfolta is that it's deliberately weak.

**Reply by Saugia** [reply]
What you describe to be a "joke" and comparisons to other ships with different gun ports are, as I keep saying, very intentional setups, and the point is those ships are meant to have those variances and be either good, average, a joke, etc. and never meant to be in different tiers in the first place.

I did not anywhere mention the Saber is going to be "a resulting stronger Korath ship", as in, too op to other Korath ships, etc. - I apologize if it was confusing, but what I meant was that the Saber is Korath-level in strength, and that in itself is against the game's principles - a ship becoming as strong as another tier in itself is a balance concern, in part due to comparison, and in part due to the ship not supposed to be there in the first place. The player should not have an ability to change the principle of a ship's relevance based on every stat - it is something that we already have the power in outfit expansions, changes, cargo, etc., and this is the full extent of what an "internal refit" is meant to be, and is balanced to be the way it is, and not more than what it is (or less). If you're able to make a build a tier higher or more relevant with what you can already do in the game, that's great! If you're not able to, that's also fine, and how it's meant to be. 

At this point, I believe everything has been said that has needed to be said. Even should ships with this given internals upgrade still not be superficially better than ships in the given faction's tier, the whole intent is these ships should not be there in the first place, and more importantly, not done so with manipulation base stats. It can come to balance concerns, and bouncing back to the other side, *this is intentional game design*. 

You seem to disagree with what we believe to be appropriate for the game, and I entirely understand what you believe would be good. Neither is necessarily a wrong option. However, we have made choices of what vanilla's design choice is based around. On the same note, you can (and probably already could have) made a plug-in to add this functionality, and you'd be able to do this as you will, and we'd still work on the game as we as developers believe the game is best made. Some day I have mind to make plug-ins for things I would have preferred been in the game as well, but I'd like to focus my energy on vanilla first and foremost.

I appreciate you suggesting something different for the game. Our response is we don't believe it is a suggestion fit for vanilla, and we've given reasons above as to why. I've decided to close the discussion so we don't spend a lot of time going in circles after replying to the suggestion.
